### ISSUE TITLE
MarkdownIt().render(src) errors on null src values

### ISSUE THREAD

**Issue by slackboxster** [markdown-it/markdown-it#324]
`MarkdownIt().render(src)` gives the following error when `src` is null:
`TypeError: Cannot read property 'replace' of undefined`

Is it within scope to check for null inputs at some point?

For my use case that would be the preferred behavior, as I am passing data into the render that may or may not be null. I can easily do a null check myself, but it would be at the very least nicer if there was a console log to give a meaningful error.

**Comment by puzrin** [comment]
There are 2 question:

- I think, if signature requires string, it's a user's responsibility to provide valid data.
- If you had no problem to find the source of error, then improvments are not necessary.

What do you think?

PS. IMHO everything is good enougth as is :)

**Comment by slackboxster** [comment]
User's responsibility makes sense to me. It would simply be my personal preference to be able to throw potentially null data at the library. But it isn't my library. ;)

I did have a bit of a challenge finding the source of the error. It took me around 30 minutes to figure it out. I had to look at the library markdown-it and trace back from the point of the error to the entry point of the code to figure out where the undefined value was coming from. So it wasn't trivial to find the error. It wasn't very difficult, but I wouldn't say it was "no problem". 

I think it would be better to either give an explicit error message saying "src cannot be null", or just return null for a null input string, as it would save the user from having to poke through other code to figure out his own issue.

I would say everything I've used in your library so far is very good. And I don't think you are unjustified in calling it good enough in this instance. I do think this one small thing could be a little better. :)

In any case, thanks for looking this over. :)
